File: erpnext/gp_erp/controllers/stock/pick_list.js
frappe.ui.form.on('Pick List Item', {
	batch_no: (frm, cdt, cdn) => {
		var d = locals[cdt][cdn];
		get_balance_qty(d.batch_no, d.warehouse, d.item_code).then(res => {
			frappe.model.set_value(cdt, cdn, "balance_qty", res);
		});
	}
});

function get_balance_qty(batch_no, warehouse, item_code) {
	return frappe.xcall('erpnext.stock.doctype.batch.batch.get_batch_qty', {
		batch_no, warehouse, item_code
	});
}

File: erpnext/gp_erp/controllers/stock/pick_list.py
import frappe
from frappe import _
from frappe.utils import flt, today

from erpnext.stock.doctype.pick_list.pick_list import PickList
from erpnext.stock.doctype.batch.batch import get_batch_qty


class PickListGP(PickList):
    def validate(self):
        self.set_missing_values()
        self.validate_for_qty()

    def set_missing_values(self):
        for d in self.locations:
            d.picked_qty = flt(d.picked_qty_view) * flt(d.conversion_factor)
